### ISSUE TITLE
Aer 0.13.1 raises an import error with Qiskit 0.44.3

### ISSUE THREAD

**Issue by t-imamichi** [Qiskit/qiskit-aer#2007]
### Informations

- **Qiskit Aer version**: qiskit==0.44.3, qiskit-terra==0.25.3, qiskit-aer==0.13.1
- **Python version**: Python 3.10.13
- **Operating system**: macOS 14.1.1

### What is the current behavior?

Importing AerSimulator causes an import error with DiagonalGate.

### Steps to reproduce the problem

```
from qiskit_aer import AerSimulator
print('success')
```

qiskit==0.44.3, qiskit-terra==0.25.3, qiskit-aer==0.13.1
```
ImportError: cannot import name 'DiagonalGate' from 'qiskit.circuit.library' (/.../lib/python3.10/site-packages/qiskit/circuit/library/__init__.py)
```

qiskit==0.44.3, qiskit-terra==0.25.3, qiskit-aer==0.13.0
```
success
```

qiskit==0.45.0, qiskit-terra==0.45.0, qiskit-aer==0.13.1
```
success
```

### What is the expected behavior?

No error with qiskit==0.44.3, qiskit-terra==0.25.3, qiskit-aer==0.13.1

**Comment by doichanj** [comment]
I made fix #2008 for this issue. I think to avoid this issue is to install QIskit 0.45.0 after installing Aer 0.13.1 currently.

**Comment by t-imamichi** [comment]
Fixed by #2008
